gwpy/timeseries/io/gwf/lalframe.py

```diff
@@ -155,6 +155,19 @@ def read(source, channels, start=None, end=None, series_class=TimeSeries,
     for name in channels:
         ts = _read_channel(stream, str(name), start=start, duration=duration)
         out[name] = series_class.from_lal(ts, copy=False)
+        tsend = ts.epoch + getattr(ts.data.data, "size", 0) * ts.deltaT
+        if (end - tsend) >= ts.deltaT:  # one (probably) sample missing
```

Comment:
Ok, I understand after reading the issue. I didn't understand the comment first time around. I wonder if it's possible to write "sometimes the dependency does not include the last sample due to rounding". I read "we know there is data missing" to mean, there is data missing on the other side, rather than we did not receive the missing data.

Reply:
I think (hope) that the comment immediately below this line explains things a bit better.